com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapterContext.java

```diff
@@ -44,14 +45,19 @@
     private RecyclableObjectPool<Long, Object> recyclableIdPool = new RecyclableObjectPool<>();
     private IVariableFormatter variableFormatter = VariableFormatterFactory.createVariableFormatter();
 
-    public DebugAdapterContext(Consumer<Messages.ProtocolMessage> messageConsumer, IProviderContext providerContext) {
+    public DebugAdapterContext(ProtocolServer server, IProviderContext providerContext) {
         this.providerContext = providerContext;
-        this.messageConsumer = messageConsumer;
+        this.server = server;
     }
 
     @Override
     public void sendEvent(DebugEvent event) {
-        messageConsumer.accept(new Messages.Event(event.type, event));
+        server.sendMessage(new Messages.Event(event.type, event));
+    }
+
+    @Override
+    public void sendRequest(Messages.Request request, int timeout, Consumer<Messages.Response> cb) {
+        server.sendRequest(request, timeout, cb);
```

Comment:
If we want to use the server to sendMessage or sendRequest, we can introduce a new interface like IProtocolServer which has members like sendMessage, sendRqeust, sendEvent.